### PR TITLE
DBT-753 DBT-752 List of roles, including registering the most basic

### PR DESCRIPTION
**Pull request by kayiganwa04**
## Context

This is a new feature for the roles task

[Task](https://www.notion.so/tianlu/Frontend-List-of-roles-including-registering-the-most-basic-a5969af972124581beed6ea402a18f4c?pvs=4)

## Test

- On the sidebar check for roles
- You can add a role
- and you can edit a role

## Platform

Google chrome

### CODE REVIEW

File: src/configs/navigation.js [hunk 1]
```diff
@@ -156,6 +156,13 @@ export default {
           to: { name: 'manage-lessons' },
           roles: 'admin'
         },
+        {
+          icon: 'mdi-clipboard-account',
+          key: '',
+          text: 'Profile',
```
Comment (abelbordonado) [inline src/configs/navigation.js:162]: 🇪🇸 Gestión de Permisos

File: src/modules/permissions/profiles-list/components/profiles-list.vue [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            <ResourceButtonAdd text-button="Crear un perfil" @click="addProfile()" />
+            <resource-button
+              icon-button="mdi-autorenew"
+              @click="resetTableOptions"
+            />
+          </template>
+        </Toolbar>
+        <resource-text-field-search
+          :search-word="store.tableOptions.content"
+          label-text-field="Buscar por nombre de la prefil."
```
Comment (abelbordonado) [inline src/modules/permissions/profiles-list/components/profiles-list.vue:22]: 🇪🇸  `Buscar por nombre del prefil.` (de la -> del)

File: src/modules/permissions/profiles-list/components/profiles-list.vue [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+          params: {
+            id: item.id
+          }
+        })
+    },
+    canEdit(profile) {
+      if (!profile) {
+        return
+      }
+      const Protected = profile.protected
```
Comment (abelbordonado) [inline src/modules/permissions/profiles-list/components/profiles-list.vue:135]: small P, `protected` 

`const {protected} = profile ; ` is the eslint preferred way

File: src/modules/permissions/profiles-list/components/profiles-list.vue [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+      return Protected === 1 ? true : false
```
Comment (abelbordonado) [inline src/modules/permissions/profiles-list/components/profiles-list.vue:137]: LOL!

`return protected === 1` is enough!  (that will be true or false already)

File: src/modules/permissions/update-profile/index.vue [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      </v-toolbar-title>
+    </v-toolbar>
+    <validation-observer ref="FormCreateProfile">
+      <section class="px-2 py-2 d-flex flex-sm-column align-center">
+        <v-row dense :style="{ width: '-webkit-fill-available' }">
+          <v-col cols="12" sm="12" md="6" lg="4">
+            <!-- Nombre -->
+            <NameFieldInput
+              v-model="name"
+              rules="required|min:3|max:25|regex:^[a-zA-Z0-9áéíóúÁÉÍÓÚñÑ _-]+$"
```
Comment (abelbordonado) [inline src/modules/permissions/update-profile/index.vue:24]: Make this constant and replace it everywhere

`regex:^[a-zA-Z0-9áéíóúÁÉÍÓÚñÑ _-]+$` 

// Const file
`const inputValidRegex = "egex:^[a-zA-Z0-9áéíóúÁÉÍÓÚñÑ _-]+$"`

// Here
`rules={`equired|min:3|max:25|${inputValidRegex}`}`

File: src/modules/permissions/update-profile/index.vue [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+      if (!status) {
+        this.$swal.fire({
+          icon: 'error',
+          toast: true,
+          title: 'Por favor, complete correctamente los campos del formulario.',
+          showConfirmButton: true,
+          confirmButtonText: 'Entendido',
+          timer: 7500
+        })
```
Comment (abelbordonado) [inline src/modules/permissions/update-profile/index.vue:112]: What about our friendly helper function?  

`Toast.error({ title: 'asdasda' })`

File: src/modules/permissions/update-profile/index.vue [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+      }
+
+      await this.$swal.fire({
+        icon: 'success',
+        toast: true,
+        title: 'Prefil Creado!',
+        showConfirmButton: true,
+        confirmButtonText: 'Entendido',
+        timer: 7500
+      })
```
Comment (abelbordonado) [inline src/modules/permissions/update-profile/index.vue:139]: What about our friendly helper function?

Toast.success({ title: 'asdasda' })

File: src/services/ProfileRepository.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+    if (response.status === 409) {
+      ResourceService.warning({
+        response,
+        title: 'Información Duplicada',
+        message: 'Ya existe un grupo con el mismo código o color'
+      })
+
+      return false
+    }
```
Comment (abelbordonado) [inline src/services/ProfileRepository.js:77]: Helpfull with copy pasting!

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

File: src/modules/permissions/profiles-list/components/profiles-list.vue (hunk 3 above)
Comment (kayiganwa04) [reply src/modules/permissions/profiles-list/components/profiles-list.vue:138]: Identifier expected. 'protected' is a reserved word in strict

File: src/modules/permissions/profiles-list/components/profiles-list.vue [hunk 9]
```diff
@@ -151,6 +176,27 @@ export default {
...
       return res
     },
+    async deleteRole(role) {
+      if (!role) {
+        return
+      }
+      const response = await Toast.dialog('ELIMINAR Role', '¿Seguro que deseas eliminar este Role? Todos los Role seran borrados del servidor y los alumnos no podrán acceder a ellos')
+
+      if (!response.isConfirmed) {
+        console.log({ response })
```
Comment (abelbordonado) [inline src/modules/permissions/profiles-list/components/profiles-list.vue:186]: 🦞

File: src/modules/permissions/profiles-list/components/profiles-list.vue [hunk 10]
```diff
@@ -22,7 +22,7 @@
         </Toolbar>
         <resource-text-field-search
           :search-word="store.tableOptions.content"
-          label-text-field="Buscar por nombre del perfil."
+          label-text-field="Buscar por nombre del prefil."
```
Comment (abelbordonado) [inline src/modules/permissions/profiles-list/components/profiles-list.vue:25]: This came back `perfil.` was right  🇪🇸

**Comment by abelbordonado** [comment]
image
Format as the other tables

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Review by abelbordonado — APPROVED** [review]
Approved all is working fine!

**Comment by abelbordonado** [comment]
Included in #21

**Pull request closed without merging**